source/Weapon.h

```diff
@@ -33,6 +35,22 @@ class Sprite;
 // copy of them, and storing them as class variables instead of in a map of
 // string to double significantly reduces access time.
 class Weapon {
+public:
+	struct Submunition{
+		Submunition() noexcept = default;
+		explicit Submunition(const Weapon *weapon, std::size_t count) noexcept
+			: weapon(weapon), count(count) {};
+		
+		const Weapon *weapon;
+		// The direction and offset at which the submunition will be fired from
+		// the original projectile. The direction and offset are relative to the
+		// point and direction at which the original projectile dies.
+		Angle facing;
+		Point offset;
+		int count;
```

Comment:
Initialization is needed for the default constructor, plus comments scoped to the respective variable:
```suggestion
		const Weapon *weapon = nullptr;
		std::size_t count = 0;
		// The angular offset from the source projectile, relative to its current facing.
		Angle facing;
		// The base offset from the source projectile's position, relative to its current facing.
		Point offset;
```